hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -703,6 +705,19 @@ CompletableFuture<ContainerCommandResponseProto> writeChunkToContainer(
       LOG.debug("Writing chunk {} length {} at offset {}",
           chunkInfo.getChunkName(), effectiveChunkSize, offset);
     }
+
+    final ChunkInfo previous = previousChunkInfo.getAndSet(chunkInfo);
+    final long expectedOffset = previous == null ? 0
+        : chunkInfo.getChunkName().equals(previous.getChunkName()) ?
```

Comment:
what's the case of the current chunk being equal to the previous chunk? Is it for chunk retries?

Reply:
https://github.com/apache/ozone/blob/200e5fe41c07ad4993fec5647d2cd8774798da7c/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java#L698

On a second thought, the chunk name will never be the same due to line 698 above.

Reply:
Anyway, let's keep the current code since it is logically correct although the chunk names seem to never be equal.